### PR TITLE
Make `SymbolAnnotation` static

### PR DESCRIPTION
**Pull request by Youssef1313**
Noticed while going through the code that this class isn't intended to be instantiated. All members in it are static, so it can be too.

### CODE REVIEW

File: src/Workspaces/SharedUtilitiesAndExtensions/Compiler/Core/Simplification/SymbolAnnotation.cs [hunk 1]
```diff
@@ -10,7 +10,7 @@ namespace Microsoft.CodeAnalysis.Simplification
     /// <summary>
     /// An annotation that holds onto information about a type or namespace symbol.
     /// </summary>
-    internal class SymbolAnnotation
+    internal static class SymbolAnnotation
```
Comment (mavasani) [inline src/Workspaces/SharedUtilitiesAndExtensions/Compiler/Core/Simplification/SymbolAnnotation.cs:13]: Is there an analyzer to flag this?

**Review by mavasani — APPROVED** [review]

File: src/Workspaces/SharedUtilitiesAndExtensions/Compiler/Core/Simplification/SymbolAnnotation.cs (hunk 1 above)
Comment (Youssef1313) [reply src/Workspaces/SharedUtilitiesAndExtensions/Compiler/Core/Simplification/SymbolAnnotation.cs:13]: @mavasani 
https://github.com/dotnet/roslyn-analyzers/blob/2928b857198370bf518b8e5647188d2a055f5928/src/NetAnalyzers/Core/Microsoft.CodeQuality.Analyzers/ApiDesignGuidelines/StaticHolderTypes.cs#L44-L52

Not sure why it's disabled by default though.

I opened https://github.com/dotnet/roslyn/pull/65830 to enable it in this repo.

**Comment by Youssef1313** [comment]
@CyrusNajmabadi Can we get this PR merged please? Thank you

**Review by CyrusNajmabadi — APPROVED** [review]

**Comment by Youssef1313** [comment]
@CyrusNajmabadi Could you restart the failing job please? Thank you

**Pull request merged**